analytics: Add summary statistic for upload/storage space in use.

laurynmm opened the pull request:
Adds the realm's used storage space for attachments to the stats view page parameters. This information is then added to the summary statistics section of the analytics page after being formatted by `stats.js`.

Part of #20162. Relevant [CZO conversation](https://chat.zulip.org/#narrow/stream/3-backend/topic/used.20upload.20space.20.20CountStat/near/1442766).

**Notes**:
- Currently, not skipping the caching of `currently_used_upload_space_bytes`, which is part of the plan (see CZO link above).
- *Testing question*: So far, I have only manually tested these changes. Is there a place where adding some automated tests makes sense? The function `currently_used_upload_space_bytes` now being called in `stats.py` has automated test coverage in `test_uploads.py`. And `stats.js` has no automated tests.
- Currently, if either there is an error getting the CountStat data or the realm was created less than 24 hours ago (i.e., `analytics_ready = False`), then this statistic is still populated if there are attachments saved in the database. Is this the behavior we want? See screenshots below.
- I used/imported the `bytes_to_size` function from `attachments_ui.js`. We already import parts of `page_params.ts` and `i18n.ts`, but I wasn't sure if it would be better to just copy the function into `stats.js` with a comment that it is duplicated from `attachments_ui.js`?

---

**Screenshots and screen captures:**

<details>
<summary>Normal realm stats page</summary>

![Screenshot from 2022-09-30 15-41-33](https://user-images.githubusercontent.com/63245456/193282699-c1a8fa73-675d-4733-be4f-14b8f4af912b.png)
</details>

<details>
<summary>New realm stats page</summary>

*Note: I have a separate task marked for improving the empty graphs generated in this situation.*
![Screenshot from 2022-09-30 16-11-10](https://user-images.githubusercontent.com/63245456/193288745-53e4121f-05aa-4c2c-88d2-8a4247c356a7.png)
</details>

<details>
<summary>Error stats page</summary>

*Note: The spinners in the graphs never stop or go away.*
![Screenshot from 2022-09-30 16-10-27](https://user-images.githubusercontent.com/63245456/193288594-e0be8f75-73a4-4d57-b1f9-473d808235a4.png)
</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

timabbott commented:
> Testing question: So far, I have only manually tested these changes. Is there a place where adding some automated tests makes sense?

Yeah, we only do manual testing for the stats page, via the Analytics realm. 

So the main thing worth doing is making sure the `populate_analytics_db` code that populates the analytics realm does __something__ to make it the case that it'll have data for this chart; probably the best option is to have it upload a single file present in the codebase in a message, or something like that?

> I used/imported the bytes_to_size function from attachments_ui.js. We already import parts of page_params.ts and i18n.ts, but I wasn't sure if it would be better to just copy the function into stats.js with a comment that it is duplicated from attachments_ui.js?

Copying is probably simplest. While we've been trying to make app modules only do something if their explicit `initialize` functions are called, I'd like to not have to think about whether modules that `attachments_ui` imports now or in the future could have side effects. 

> Currently, if either there is an error getting the CountStat data or the realm was created less than 24 hours ago (i.e., analytics_ready = False), then this statistic is still populated if there are attachments saved in the database. Is this the behavior we want? See screenshots below.

That's probably fine. I think the direction we're likely to go for too-new realms is more likely to involve replacing that state with a simpler error page that just tells you stats aren't available yet but you can see what they might look like via a public page like chat.zulip.org/stats.

laurynmm commented:
> So the main thing worth doing is making sure the `populate_analytics_db` code that populates the analytics realm does **something** to make it the case that it'll have data for this chart; probably the best option is to have it upload a single file present in the codebase in a message, or something like that?

That makes sense. I've added some code to `populate_analytics_db` that uses `static/images/test-images/checkbox.png` (which is also used in `populate_db`) to create an attachment linked to Shylock's account. It isn't linked to / claimed by a message, but it is still in the database, so it populates that statistic (see screenshot below).

> Copying is probably simplest. While we've been trying to make app modules only do something if their explicit `initialize` functions are called, I'd like to not have to think about whether modules that `attachments_ui` imports now or in the future could have side effects.

Yeah, that was my thought / concern too. I updated the PR to copy the `bytes_to_size` function with a comment that it's the same as in `attachments_ui.js`.

> I think the direction we're likely to go for too-new realms is more likely to involve replacing that state with a simpler error page that just tells you stats aren't available yet but you can see what they might look like via a public page like chat.zulip.org/stats.

Cool. I've got that down as a follow-up (either as a task for me or an issue to put up).

---

**Screenshot**:

<details>
<summary>Summary stats in Analytics realm with checkbox emoji as attachment</summary>

![Screenshot from 2022-10-07 15-31-28](https://user-images.githubusercontent.com/63245456/194584074-e99345e9-2c58-4f43-bce0-1c91aec53ffc.png)
</details>

timabbott commented:
Merged, after changing "Storage space in use" => "File storage in use" to better match how we describe that in the /plans page; I think it's also a bit clearer that we're not counting the size of the messages themselves.
